### DISCUSSION TITLE
How to use Dark Mode when not all pages are using SSR?

### DISCUSSION THREAD

**Discussion by amirhhashemi** [chakra-ui/chakra-ui#5366] [Q&A] [answered]
I have a next.js app and I'm trying to migrate from tailwindcss to chakra. some of the pages are using SSR. I followed the [instructions in the docs](https://chakra-ui.com/docs/features/color-mode#add-colormodemanager-optional-for-ssr) about using `Color Mode` in server-side rendered pages but I have one issue.

let's say the default theme is dark. when users visit an SSR page for the first time and change the theme to light a cookie will be set that tells other SSR pages the user theme. But non-SSR pages are using `localStorageManager` so they ignore the cookie and use the default theme and users that visit the page for the first time see the default theme instead of the one they set. How could I solve that?

I exactly followed [the docs](https://chakra-ui.com/docs/features/color-mode#add-colormodemanager-optional-for-ssr) so I use the reusable `getServerSideProps` function that was suggested there in SSR pages.

**Comment by ghost** [comment, marked as answer]
Hey, I am currently using this approach and it seems to work fine:

```
// _app.tsx 

const { cookies } = pageProps;

const colorModeManager = cookieStorageManager(
    typeof cookies === "string"
      ? cookies
      : typeof document !== "undefined"
      ? document.cookie
      : ""
  );
```

So I don't ever use the localStorageManager. I don't know why the documentation does it. I doesn't make sense to me.

**Reply by amirhhashemi** [reply]
Thank you so much. That seems to work for me too.

**Reply by TIMMLOPK** [reply]
> Hey, I am currently using this approach and it seems to work fine:
> 
> ```
> // _app.tsx 
> 
> const { cookies } = pageProps;
> 
> const colorModeManager = cookieStorageManager(
>     typeof cookies === "string"
>       ? cookies
>       : typeof document !== "undefined"
>       ? document.cookie
>       : ""
>   );
> ```
> 
> So I don't ever use the localStorageManager. I don't know why the documentation does it. I doesn't make sense to me.

It make usecolorVaule don’t work as usual

**Reply by ghost** [reply]
What's different?

**Reply by TIMMLOPK** [reply]
> What's different?

It will be this, Are you have solutions 
![321EA94E-EB0D-4015-928A-32EEE1F8FAEA](https://user-images.githubusercontent.com/72869479/155934028-2ea956ab-a5e3-4606-b70e-8f66105ca1fe.jpeg)
